.github/workflows/build.yml

```diff
@@ -0,0 +1,260 @@
+name: Build
+on: [push]
+     
+jobs:    
+  fe-deps:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup Node.js 
+        uses: actions/setup-node@v2
+        with:
+          node-version: 12
+      - name: Install deps
+        run: yarn install
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+      - uses: actions/cache@v2
+        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+  
+  be-deps:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Install Java 11
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: '11'
+      - name: Install clojure tools
+        uses: DeLaGuardo/setup-clojure@3.2
+        with:
+          cli: 1.10.3.822
+          lein: latest
+      - name: Get project dependencies
+        run: lein deps :tree
+      - name: Cache local Maven repository
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+    
+  check-migrations:
+    runs-on: ubuntu-latest
+    needs: be-deps
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Cache local Maven repository
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - name: Install Java 11
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: '11'
+      - name: Install clojure tools
+        uses: DeLaGuardo/setup-clojure@3.2
+        with:
+          cli: 1.10.3.822
+          lein: latest
+      - name: Get project dependencies
+        run: lein deps :tree
+      - name: Verify Liquibase migrations
+        run: ./bin/lint-migrations-file.sh
+
+  fe-tests:
+    runs-on: ubuntu-latest
+    needs: [fe-deps, be-deps]
+    strategy:
+      matrix:
+        tests: [test-timezones, test-unit, test-cljs]
```

Comment:
The first two should be covered by PR #15507. The last one likely requires its own yml file so we can specify a good path condition (inside `on.paths` clause).